### PR TITLE
Ci/in flight proposals check

### PR DESCRIPTION
**Pull request by anajuliabit**

### CODE REVIEW

File: script/ValidateActiveProposals.s.sol [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+contract ValidateActiveProposals is Script, Test, ChainIds {
+    /// @notice addresses contract
+    Addresses addresses;
+
+    /// @notice fork ID for moonbeam
+    uint256 public moonbeamForkId =
+        vm.createFork(
+            vm.envOr("MOONBEAM_RPC_URL", string("moonbeam")),
+            6737902
```
[the remark this answers is not in the document]
Comment (anajuliabit) [reply script/ValidateActiveProposals.s.sol:22]: that was intentional for testing purposes, this way I don't need to recreate proposals every 5 minutes to have proposals active on testnets

**Review by prateek105 — DISMISSED** [review]
lgtm, just one query: Why are we checking active proposals, aren't they already tested. Is it because anyone can add a proposal? @anajuliabit

**Comment by anajuliabit** [comment]
@prateek105 yes and also to add an extra layer check, we can't guarantee that the calldata posted on chain was the same validated by the testBranchProposals test

File: src/proposals/proposalTypes/ProposalChecker.sol [hunk 2]
```diff
@@ -201,11 +201,13 @@ abstract contract ProposalChecker is ChainIds {
...
+
     function getTargetsPayloadsValues(
         Addresses addresses
     )
         public
         view
         virtual
-        returns (address[] memory, uint256[] memory, bytes[] memory);
+        returns (address[] memory, uint256[] memory, bytes[] memory)
+    {}
```
[the remark this answers is not in the document]
Comment (anajuliabit) [reply src/proposals/proposalTypes/ProposalChecker.sol:212]: I did that to not have to override on integration test but will override then

File: src/utils/Address.sol [hunk 3]
```diff
@@ -0,0 +1,313 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+
+/**
+ * @dev Collection of functions related to the address type
```
[the remark this answers is not in the document]
Comment (anajuliabit) [reply src/utils/Address.sol:5]: where should I add the toBytes function then?

**Pull request merged**